networkit/cpp/io/NetworkitBinaryReader.cpp

```diff
@@ -102,46 +98,77 @@ Graph NetworkitBinaryReader::read(const std::string& path) {
 
 	// Read adjacency lists.
 	const char *adjIt = mmfile.cbegin() + header.offsetAdjLists;
-
+	const char *transpIt = mmfile.cbegin() + header.offsetAdjTranspose;
+	uint64_t adjListSize;
+	memcpy(&adjListSize, adjIt + (chunks -1) * sizeof(uint64_t), sizeof(uint64_t));
+	uint64_t transposeListSize;
+	memcpy(&transposeListSize, transpIt + (chunks -1) * sizeof(uint64_t), sizeof(uint64_t));
+
+	if(!directed) {
+		assert(adjListSize == transposeListSize);
+	} 
+	G.setEdgeCount(unsafe, adjListSize);
+	
+	uint64_t selfLoops = 0;
```

Comment:
Make atomic as concurrent write are possible. Alternatively return number of local self-loops from `constructGraph` and use omp reduction.

Reply:
Oh, that's correct. Using a `std::atomic<count>` variable here, with a single `.fetch_add(<number of self loops in chunk>, std::memory_order_relaxed);` looks like the best option here. That will be much faster than a OMP reduction (and also less complex).